cpp/src/arrow/compute/exec/hash_join_node.cc

```diff
@@ -516,6 +684,125 @@ class HashJoinNode : public ExecNode {
 
   const char* kind_name() const override { return "HashJoinNode"; }
 
+  Status OnBuildSideBatch(size_t thread_index, ExecBatch batch) {
+    std::lock_guard<std::mutex> guard(build_side_mutex_);
+    accumulators_[1].InsertBatch(std::move(batch));
```

Comment:
Can we use constants instead of `1`?  Something like `accumulators_[kBuildSide]`.  Or, if there is only ever going to be two, can we just have `build_accumulator` and `probe_accumulator`?

Reply:
Done